app/services/creator_analytics/churn.rb

```diff
@@ -0,0 +1,407 @@
+# frozen_string_literal: true
+
+# Calculates customer churn rate metrics for subscription products using Elasticsearch.
+#
+# Churn rate formula (per Stripe):
+#   (Churned Customers / Total Base Customers) × 100
+#
+# Where Total Base = Active at Start + New During Period
+#
+# @see https://stripe.com/resources/more/monthly-churn-101
+class CreatorAnalytics::Churn
+  include ActiveModel::Validations
+
+  DEFAULT_START_DATE_OFFSET = 1.month
+
+  validates :end_date, comparison: { greater_than_or_equal_to: :start_date }
+
+  attr_reader :start_date, :end_date, :user, :products
+
+  def initialize(user:, start_date: nil, end_date: nil, params: {}, products: nil)
+    @user = user
```

Comment:
`user` is quite generic, and can represent a _seller_ or a _buyer_. It's better to have the code self-document the type of user referenced.

```suggestion
    @seller = seller
```